server/src/main/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorService.java

```diff
@@ -279,11 +281,11 @@ public HealthIndicatorResult calculate(boolean explain) {
                 tier -> tier,
                 tier -> new UserAction.Definition(
                     "increase_tier_capacity_for_allocations_" + tier,
-                    "Elasticsearch isn't allowed to allocate some shards from these indices to any of the nodes in their data tiers "
+                    "Elasticsearch isn't allowed to allocate some shards from these indices to any of the nodes in the desired data tier "
```

Comment:
If the user is using multiple tiers, this information might not be enough. Could we specify which tier we're talking about?

Reply:
We discussed off issue and decided to move this to the next edit PR.